src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -2043,6 +2049,14 @@ private void WriteRedMinimalFactory(Node nd, bool withStringNames = false)
 
             this.WriteLine();
 
+            var hasOptional = minimalFactoryfields.Any(f => !IsRequiredFactoryField(nd, f));
+            var hasAttributeOrModifiersList = nd.Fields.Any(f => IsAttributeOrModifiersList(f));
+
+            if (hasOptional && hasAttributeOrModifiersList)
+            {
+                WriteLine("#pragma warning disable RS0027");
```

Comment:
Why is disabling this warning necessary here?

Reply:
Ah.  So we have an annoyin interaction between our public-api checker and how we decied to generate nodes initially.  This leads to this necessity.  Specifically, we have a public-API checker rule that says:

"if you have an overload with optional parameters, you cannot have overlaods with non-optional parameters that have more parameters".  or, put more simply, "only one overload can have optional parameters and it must be the one with the most parameters".

*However*, this rule does not apply to the syntax model where we violate this heavily.  i.e. we *already* have overloads like:

```c#
public static BlockSyntax Block(SyntaxList<StatementSyntax> statements = default(SyntaxList<StatementSyntax>))
public static BlockSyntax Block(SyntaxToken openBraceToken, SyntaxList<StatementSyntax> statements, SyntaxToken closeBraceToken)
```

So why was this not a problem already?  Because the public-api checker explicitly suppresses itself *if* the apis are already shipped, throwing its hands up in the air and basically saying welp... nothing we can do about this now for this API."

--
In this case though, we are adding some new overloads for these methods.  for example:

```c#
pu3blic static BlockSyntax Block(SyntaxList<AttributeListSyntax> attributeLists, SyntaxToken openBraceToken, SyntaxList<StatementSyntax> statements, SyntaxToken closeBraceToken)
```

So the api checker doesn't suppress itself here.  

This puts us in a bind where effectively the api checker is saying: you cannot evolve these specific nodes to add more children, something which is actually ok (if we do it carefully like here).  

So the approach is to just suppress in this case because we know that adding this stuff to these nodes isn't breaking any compat, it's just to suppress hte overzealous api checker which should apply to new APIs but which needs to be ok with these changes for these already shipped APIs.

Reply:
Would it make sense to open a tracking issue for preventing these warnings from being produced at all on the violating overload with too many parameters is unshipped, and the related overload with optional parameters is already shipped?

Reply:
yes, that would likely make sense.  i will do that.

Reply:
https://github.com/dotnet/roslyn-analyzers/issues/2879

Thanks!